src/udev/udevd.c

```diff
@@ -343,30 +343,52 @@ static int worker_lock_block_device(struct udev_device *dev, int *ret_fd) {
          * udev has finished its event handling.
          */
 
-        if (streq_ptr(udev_device_get_action(dev), "remove"))
+        r = sd_device_get_property_value(dev, "ACTION", &val);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to get the value of property 'ACTION': %m");
+
+        if (streq(val, "remove"))
                 return 0;
 
-        if (!streq_ptr("block", udev_device_get_subsystem(dev)))
+        r = sd_device_get_subsystem(dev, &val);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to get subsystem: %m");
+
+        if (!streq(val, "block"))
                 return 0;
 
-        sysname = udev_device_get_sysname(dev);
-        if (startswith(sysname, "dm-") ||
-            startswith(sysname, "md") ||
-            startswith(sysname, "drbd"))
+        r = sd_device_get_sysname(dev, &val);
+        if (r < 0)
+                return log_device_debug_errno(dev, r, "Failed to get sysname: %m");
+
+        if (startswith(val, "dm-") ||
+            startswith(val, "md") ||
+            startswith(val, "drbd"))
```

Comment:
(not directly on topic, but I think we should probably add STARTSWITH_SET() to make this prettier, similar to how we already have PATH_STARTSWITH_SET(), STR_IN_SET()…)